pkg/persistence/object_store.go

```diff
@@ -235,6 +235,16 @@ func (s *objectBackupStore) ListBackups() ([]string, error) {
 		// each of those off to get the backup name.
 		backupName := strings.TrimSuffix(strings.TrimPrefix(prefix, s.layout.subdirs["backups"]), "/")
 
+		// if a bucket is versioned, the s3 folder still exists, even if all s3 objects have been deleted.
```

Comment:
When talking about S3, is it the AWS S3 storage or all the S3-compatible storage?
I'm not sure whether this is for all S3-compatible storage.

Reply:
Right. Most S3 implementations support versioning - and some maybe do not. 
I think velero should be able to work on versioned and non versioned buckets, without producing endless error messages on versioned buckets (but still does it's job). 

This PR only disables the error messages, if velero is using a versioned bucket.  
 
BTW: we currently use this PR in a self created docker image based on version 1.12.0 based in production without any troubles.

Reply:
It makes sense to ignore this case, but maybe we should reword the comments because we cannot ensure this scenario is caused by bucket versioning, since we don't even know whether the bucket versioning is enabled or not. That is one possible scenario.